Rubberduck.Parsing/Symbols/DeclarationFinder.cs

```diff
@@ -77,64 +82,61 @@ protected virtual void ExecuteCollectionConstructionActions(List<Action> collect
             collectionConstructionActions.ForEach(action => action.Invoke());
         }
 
-        private List<Action> CollectionConstructionActions(IReadOnlyList<Declaration> declarations, IEnumerable<IAnnotation> annotations, IReadOnlyList<UnboundMemberDeclaration> unresolvedMemberDeclarations)
+        private List<Action> CollectionConstructionActions(IReadOnlyList<Declaration> declarations, IEnumerable<IAnnotation> annotations, 
+            IReadOnlyList<UnboundMemberDeclaration> unresolvedMemberDeclarations)
         {
-            var actions = new List<Action>();
+            var actions = new List<Action>
+            {
+                () =>
+                    _unresolved = unresolvedMemberDeclarations
+                        .ToList(),
+                () =>
+                    _annotations = annotations
+                        .GroupBy(node => node.QualifiedSelection.QualifiedName)
+                        .ToDictionary(),
+                () =>
+                    _declarations = declarations
+                        .GroupBy(item => item.QualifiedName.QualifiedModuleName)
+                        .ToDictionary(),
+                () =>
+                    _declarationsByName = declarations
+                        .GroupBy(declaration => declaration.IdentifierName.ToLowerInvariant())
+                        .ToDictionary(),
+                () =>
+                    _declarationsBySelection = declarations
+                        .Where(declaration => declaration.IsUserDefined)
+                        .GroupBy(GetGroupingKey)
+                        .ToDictionary(),
+                () =>
+                    _referencesBySelection = declarations
+                        .SelectMany(declaration => declaration.References)
+                        .GroupBy(
+                            reference => new QualifiedSelection(reference.QualifiedModuleName, reference.Selection))
+                        .ToDictionary(),
+                () =>
+                    _parametersByParent = declarations
+                        .Where(declaration => declaration.DeclarationType == DeclarationType.Parameter)
+                        .Cast<ParameterDeclaration>()
+                        .GroupBy(declaration => declaration.ParentDeclaration)
+                        .ToDictionary(),
+                () =>
+                    _userDeclarationsByType = declarations
+                        .Where(declaration => declaration.IsUserDefined)
+                        .GroupBy(declaration => declaration.DeclarationType)
+                        .ToDictionary(),
+                () =>
+                    _referencesByModule = declarations
+                        .SelectMany(declaration => declaration.References)
+                        .GroupBy(reference =>
+                            Declaration.GetModuleParent(reference.ParentScoping).QualifiedName.QualifiedModuleName)
+                        .ToDictionary(),
+                () =>
```

Comment:
I think the original code looked a bit more structured, but this is OK with me as well.

Reply:
That was an R# oops - I hit "fix in file" instead of "fix" somewhere along the line.  When I noticed it, I didn't have any strong feelings one way or the other so I just left it.  I can revert changes to that function if you'd prefer.

Reply:
I guessed it was that. As already stated, I am OK with the new version.